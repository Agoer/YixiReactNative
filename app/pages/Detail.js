import React, { Component } from 'react';
import {
    View,
    Text,
    Image,
    StyleSheet,
    Dimensions,
    TouchableHighlight
} from 'react-native';
import Spinner from 'react-native-spinkit';
import ParallaxView from 'react-native-parallax-view';
import {fetchLectureItem} from '../api';
import global_style from '../global_style';
const ScreenWidth = Dimensions.get('window').width;
const styles = StyleSheet.create({
  container : {
     flex: 1,
  },
  albumDetail : {
    flex : 1,
  },
  albumLecturer : {
     flex : 1,
     alignItems : 'center',
     justifyContent : 'center',
     paddingVertical : 60,
     marginHorizontal : 12,
     borderBottomWidth :  StyleSheet.hairlineWidth,
     borderBottomColor : '#999'
  },
  lecturerAvatar : {
     position : 'absolute',
     height : 80,
     width : 80,
     borderRadius : 41,
     borderWidth : 2,
     borderColor : '#fff',
     backgroundColor : '#fff',
     zIndex : 5,
     top : -40
  },
  lecturetNickName : {
     fontSize : 20,
     fontWeight : '200'
  },
  lecturetDesc : {
    fontSize : 14,
    fontWeight : '100',
    marginTop : 16,
    color : '#999',
  },
  albumLecture : {
     alignItems : 'center',
     paddingVertical : 32
  },
  lectureTitle : {
     fontSize : 26,
     color : '#cc3434',
     fontWeight : '300'
  },
  lectureTime : {
     marginVertical : 16,
     fontSize : 14,
     color : '#999',
     fontWeight : '100'
  },
  lectureDescContainer : {
     alignSelf : 'flex-start',
     marginHorizontal : 12,
     paddingLeft : 14,
     borderLeftColor : '#cc3434',
     borderLeftWidth : 4,
  },
  lectureDesc : {
     fontSize : 18,
     fontWeight : '100',
     lineHeight : 24,
     color : '#666'
  },
  albumVideo : {
     flex : 1,
     height : 150,
     alignItems : 'center',
     backgroundColor : '#fefefe'
  },
  videoBg : {
     resizeMode : 'cover',
     height : 180,
     width : ScreenWidth,
     zIndex : 1
  },
  videoBtn : {
     position : 'absolute',
     width : 50,
     height : 50,
     top : 65,
     zIndex : 5,
     backgroundColor : '#fff',
     borderRadius : 25
  },
  albumShortContent : {
     flex : 1,
     marginHorizontal : 12,
     marginVertical : 32,
  },
  purecontent : {
     fontSize : 18,
     fontWeight : '100',
     marginVertical : 24,
     lineHeight : 24,
     color : '#666'
  },
  navigateBtn : {
    borderWidth : 1,
    borderColor : '#cc3434',
    borderRadius : 6,
    width : 180,
    height : 40,
    alignItems : 'center',
    justifyContent : 'center',
    alignSelf : 'center',
  },
  navigateText : {
    color : '#cc3434',
    fontSize : 22
  }
})
export default class Detail extends Component{
  constructor(props){
    super(props);
    this.state = {
      lectureData : [],
      DataIdArr : [],
      loadingState : true
    }
    this.setState = this.setState.bind(this);
    this.fetchData = this.fetchData.bind(this);
    this.changeState = this.changeState.bind(this);
    this._onTouchEnd = this._onTouchEnd.bind(this);
  }
  async fetchData(lid) {
     return await fetchLectureItem(lid);
  }
  changeState(lid = 420) {
      this.fetchData(lid).then(result => {
          let tmpData = this.state.lectureData;
          let tmpDataArr = this.state.DataIdArr;
          if(tmpDataArr.indexOf(result.data.id) === -1){
            tmpData.push(result);
            tmpDataArr.push(result.data.id);
            this.setState({
              lectureData : [...new Set(tmpData)],
              loadingState : false,
              DataIdArr : tmpDataArr
            });

          }
      })
  }
  _onTouchEnd(e, state, context) {
      let { params } = this.props.navigation.state;
//      alert(state.index);
      this.changeState(419);
  }
  componentDidMount() {
    let { params } = this.props.navigation.state;
    this.changeState(params.lecturesId);
  }
  render() {
     const { params } = this.props.navigation.state;
     let LectureInfo = this.state.lectureData.data;

     return this.state.loadingState ?
                <View style={global_style.loadingContainer}>
                   <Spinner isVisible={true}
                            size={50}
                            type='ThreeBounce'
                            color='#cc3434'/>
                </View>
           : this.state.lectureData.length > 0 ?
                    <LectureItem LectureInfo={this.state.lectureData[0].data} {...this.props}/>
           : <TouchableHighlight style={global_style.hintContainer} onPress={ this.changeState }>
                <Text style={global_style.hintText}>网络异常，请点击重新加载</Text>
             </TouchableHighlight>
  }
}


class LectureItem extends Component {
    constructor(props) {
       super(props);
    }
    render() {
       let LectureInfo = this.props.LectureInfo;
       return <ParallaxView
             backgroundSource={{url : LectureInfo.lecturer.background}}
             windowHeight={180}
         >
           <View style={styles.albumDetail}>
                <View style={styles.albumLecturer} >
                     <Image source={{ url : LectureInfo.lecturer.pic}} style={styles.lecturerAvatar}/>
                     <Text style={styles.lecturetNickName}>{LectureInfo.lecturer.nickname}</Text>
                     <Text style={styles.lecturetDesc}>{LectureInfo.lecturer.desc}</Text>
                </View>
                <View style={styles.albumLecture}>
                     <Text style={styles.lectureTitle}>{LectureInfo.title}</Text>
                     <Text style={styles.lectureTime}>{LectureInfo.time + ' ' + LectureInfo.site}</Text>
                     <View style={styles.lectureDescContainer}>
                       <Text style={styles.lectureDesc}>{LectureInfo.desc}</Text>
                     </View>
                </View>
                <View style={styles.albumVideo}>
                     <Image source={{ url : LectureInfo.cover }} style={styles.videoBg} />
                     <Image source={ require('../images/play.png') } style={styles.videoBtn} />
                </View>
                <View style={styles.albumShortContent}>
                    <Text style={styles.purecontent}>{LectureInfo.purecontent.slice(0,100)}</Text>
                    <TouchableHighlight onPress={() => this.props.navigation.navigate('PureContent', {author: LectureInfo.lecturer.nickname, title: LectureInfo.title, content: LectureInfo.purecontent})}>
                        <View style={styles.navigateBtn}><Text style={styles.navigateText}>完整演讲稿</Text></View>
                    </TouchableHighlight>
                </View>
           </View>
         </ParallaxView>
    }
}
