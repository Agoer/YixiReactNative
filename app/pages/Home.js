import React, { Component } from 'react';
import {
    View,
    Text,
    Image,
    StyleSheet,
    Dimensions,
    TouchableHighlight
} from 'react-native';
import global_style from '../global_style';
import {fetchHome} from '../api';
import Spinner from 'react-native-spinkit';
import Swiper from 'react-native-swiper';
import ParallaxView from 'react-native-parallax-view';

const ScreenHeight = Dimensions.get('window').height;
const ScreenWidth = Dimensions.get('window').width;

const styles = StyleSheet.create({
  container : {
     flex: 1,
  },
  albumHeader : {
     flex : 1,
     paddingHorizontal : 24,
     paddingBottom : 32,
     justifyContent : 'flex-end'
  },
  albumTitle : {
     color : '#fff',
     fontSize : 26,
     fontWeight : '600'
  },
  albumIntro : {
     marginTop : 26,
     lineHeight : 24,
     color : '#fff',
     fontSize : 16,
     fontWeight : '100',
  },
  albumDetail : {
     backgroundColor : '#222',
     paddingHorizontal : 16,
     paddingVertical : 25
  },
  albumContent : {
    color : '#fff',
    fontSize : 18,
    lineHeight : 26,
    fontWeight : '200'
  },
  lectureLink : {
     justifyContent : 'center',
     marginTop : 36,
     height : 120,
     width : ScreenWidth - 32,
     flexWrap : 'nowrap',
     borderRadius : 8
  },
  lectureBg : {
     position : 'absolute',
     top : 0,
     left : 0,
     zIndex : 1,
     height : 120,
     width : ScreenWidth - 32,
     borderRadius : 8
  },
  lectureDesc : {
     flex : 1,
     zIndex : 3,
     paddingLeft : 10,
     justifyContent : 'center',
     borderRadius : 4,
     backgroundColor : 'rgba(0, 0, 0, 0.4)'
  },
  descTitle : {
     color : '#fff',
     fontSize : 18,
     fontWeight : '400'
  },
  descLecturer : {
     color : '#ccc',
     fontWeight : '100',
     fontSize : 14,
     marginTop : 8
  },
  linkIcon : {
     position : 'absolute',
     right : 10,
     top : 44,
     height : 12,
     width : 12,
     zIndex : 3
  }
})

export default class Home extends Component{
  constructor(props) {
    super(props);
    this.state = {
      albumList : null,
      loadingState : true,
      loadCount : 8
    }
    this.fetchData = this.fetchData.bind(this);
    this._onMomentumScrollEnd = this._onMomentumScrollEnd.bind(this);
    this.renderList = this.renderList.bind(this);
  }
  async fetchData() {
    let album = await fetchHome();
     this.setState({
       albumList : album,
       loadingState : false
     })
  }
  _onMomentumScrollEnd(e, state, context) {
    state.index === this.state.loadCount ? this.setState({
         loadCount : this.state.loadCount + 8
    }) : '';
  }
  componentDidMount() {
     this.fetchData();
  }
  renderList() {
     return this.state.loadingState ?
                <View style={global_style.loadingContainer}>
                   <Spinner isVisible={true}
                            size={50}
                            type='ThreeBounce'
                            color='#cc3434'/>
                </View>
           : this.state.albumList.data ?
                <Swiper loop={false} onMomentumScrollEnd={this._onMomentumScrollEnd} showsPagination={false}>
                   {Array.prototype.filter.call(this.state.albumList.data, (item, k) =>
                       k <= this.state.loadCount
                   ).map( (item, k) =>
                       <AlbumItem album={item} loopKey={k} key={item.id} {...this.props}/>
                   )}
                </Swiper>
           : <TouchableHighlight style={global_style.hintContainer} onPress={ this.fetchData }>
                <Text style={global_style.hintText}>网络异常，请点击重新加载</Text>
             </TouchableHighlight>
  }
  render() {
     return (
        <View style={styles.container}>
            {this.renderList()}
        </View>
     )
  }
}
class AlbumItem extends Component {
  constructor(props){
    super(props);
  }
  render() {
    let bg = this.props.album ? this.props.album.background.replace(".1536x1000","") : "";
    let linkBg = this.props.album ? this.props.album.lectures[0].lecturer.background.replace(".1242x505","") : "";
    return <ParallaxView
          backgroundSource={{url : bg}}
          windowHeight={ScreenHeight}
          header={(
              <View style={styles.albumHeader}>
                  <Text style={styles.albumTitle}>{this.props.album ? this.props.album.title : ""}</Text>
                  <Text style={styles.albumIntro}>{this.props.album ? this.props.album.desc : ""}</Text>
              </View>
          )}
      >
        <View style={styles.albumDetail}>
            <Text style={styles.albumContent}>{this.props.album ? this.props.album.purecontent : ""}</Text>
            <TouchableHighlight onPress={ () => this.props.navigation.navigate('Detail', {lectures : this.props.album, lecturesId : this.props.album.lectures[0].id, loopKey : this.props.loopKey}) }>
                <View style={styles.lectureLink}>
                    <Image source={{ url : linkBg }} style={styles.lectureBg}  resizeMode='cover'/>
                    <View style={styles.lectureDesc}>
                        <Text style={styles.descTitle}>{this.props.album.lectures[0].title}</Text>
                        <Text style={styles.descLecturer}>{ this.props.album.lectures[0].lecturer.nickname}</Text>
                    </View>
                    <Image source={require('../images/icon-link.png')} style={styles.linkIcon}/>
                </View>
            </TouchableHighlight>
        </View>
      </ParallaxView>;
  }
}
