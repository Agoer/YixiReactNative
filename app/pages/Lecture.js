import React, { Component} from 'react';
import PropTypes from 'prop-types';
import {
    View,
    Text,
    Image,
    StyleSheet,
    ScrollView,
    TouchableHighlight
} from 'react-native';
import Spinner from 'react-native-spinkit';
import ScrollableTabView, { ScrollableTabBar, } from 'react-native-scrollable-tab-view';

import global_style from '../global_style';
import {fetchLecture} from '../api';
const styles = StyleSheet.create({
    container : {
        flex: 1,
        backgroundColor : '#fff'
    },
    navigate : {
        position : 'absolute',
        right : 12,
        top : 0,
        width : 240,
        height : 60,
        alignItems : 'center',
        justifyContent : 'flex-end',
        flexDirection : 'row'
    },
    navigateBtn : {
        height : 28,
        width : 48,
        marginLeft : 10,
        marginTop : 28,
        borderRadius : 4,
        justifyContent : 'center',
        alignItems : 'center',
        backgroundColor: '#ddd'
    },
    BtnActived : {
        backgroundColor : '#cc3434'
    },
    BtnText : {
        fontSize : 14,
        fontWeight : '100',
        color : '#333'
    },
    BtnTextActived : {
        color : '#fff'
    },
    lecturerContainer : {
        marginTop : 80,
        flex : 1,
        backgroundColor : '#fff'
    },
    barTextStyle : {
        fontWeight : '100',
        fontSize : 14
    },
    underlineStyle : {
        borderTopLeftRadius : 2,
        borderTopRightRadius : 2,
    },
    lectureListItem : {
        backgroundColor : '#f8f8f8',
        borderBottomWidth : StyleSheet.hairlineWidth,
        borderBottomColor : '#ddd',
        paddingVertical : 8,
        flexDirection : 'row'
    },
    itemCover : {
        width : 100,
        height : 68,
        marginHorizontal : 10,
        resizeMode : 'contain'
    },
    itemDetail : {
        flex : 1,
        flexDirection : 'column'
    },
    detailHeader : {
        flex : 1,
        height : 18,
        fontSize : 16,
        fontWeight : '400'
    },
    detailAuthor : {
        flex : 1,
        flexDirection : 'row',
        justifyContent : 'flex-start',
        alignItems : 'center',
        height : 16,
    },
    authorText : {
        fontSize : 13,
        fontWeight : '200',
        color : '#999',
        marginRight : 8
    },
    detailInfo : {
        flex : 1,
        flexDirection : 'row'
    },
    InfoItem : {
        flex : 1,
        flexDirection : 'row',
        justifyContent : 'flex-start',
        alignItems : 'center',
        alignSelf : 'flex-end'
    },
    InfoIcon : {
        width : 14,
        height : 14,
        resizeMode : 'contain'
    },
    InfoText : {
        fontSize : 13,
        marginLeft : 4,
        color : '#999',
        fontWeight : '200'
    }
});
export default class Lecture extends Component{
    static propTypes = {
        navigation: PropTypes.any
    };
    constructor(props) {
        super(props);
        this.state = {
            albumList : [],
            loadingState : true,
            lectureType : 'date'
        };
        this.fetchData = this.fetchData.bind(this);
        this.renderList = this.renderList.bind(this);
        this.renderLectures = this.renderLectures.bind(this);
    }
    async fetchData(api_type='date') {
        this.setState({
            lectureType : api_type,
            loadingState : true,
        });
        let album = await fetchLecture(api_type);
        this.setState({
            albumList : album,
            loadingState : false
        });
    }
    componentDidMount() {
        this.fetchData();
    }
    renderLectures(lectures) {
        return (
            lectures.map( (ele, ind) =>
                <TouchableHighlight key={`lectures-${ind}`} onPress={ () => this.props.navigation.navigate('Detail',{lectures : ele, lecturesId : ele.id }) }>
                    <View style={styles.lectureListItem}>
                        <Image source={{ url : ele.cover }} style={styles.itemCover} />
                        <View style={styles.itemDetail}>
                            <Text style={styles.detailHeader}>{ele.title}</Text>
                            <View style={styles.detailAuthor}><Text style={styles.authorText}>{ele.lecturer.nickname}</Text><Text style={styles.authorText}>{ele.time}</Text></View>
                            <View style={styles.detailInfo}>
                                <View style={styles.InfoItem}>
                                    <Image source={require('../images/ico-info-play.png')} style={styles.InfoIcon}/>
                                    <Text style={styles.InfoText}>{ele.viewnum}</Text>
                                </View>
                                <View style={styles.InfoItem}>
                                    <Image source={require('../images/ico-info-like.png')} style={styles.InfoIcon}/>
                                    <Text style={styles.InfoText}>{ele.likenum}</Text>
                                </View>
                                <View style={styles.InfoItem}>
                                    <Image source={require('../images/ico-info-comment.png')} style={styles.InfoIcon}/>
                                    <Text style={styles.InfoText}>{ele.cmtnum}</Text>
                                </View>
                            </View>
                        </View>
                    </View>
                </TouchableHighlight>
            )
        );
    }
    renderList() {
        let albumList = this.state.albumList.data;
        if(this.state.lectureType === 'date') {
            return (
                <ScrollableTabView
                    tabBarActiveTextColor='#cc3434'
                    tabBarTextStyle = {styles.barTextStyle}
                    tabBarUnderlineStyle = { styles.underlineStyle}
                    renderTabBar={() => <ScrollableTabBar />}
                >
                    {albumList.map( (ele, ind) =>
                        <ScrollView tabLabel={ele.date} key={`date-${ind}`}>
                            {this.renderLectures(ele.lectures)}
                        </ScrollView>
                    )}
                </ScrollableTabView>
            );
        } else {
            return (
                <ScrollView>{this.renderLectures(albumList)}</ScrollView>
            );
        }
    }
    render() {
        return (
            <View style={styles.container}>
                <View style={styles.navigate}>
                    <TouchableHighlight onPress={()=>this.fetchData('date')} style={[styles.navigateBtn, this.state.lectureType === 'date' && styles.BtnActived]}>
                        <Text style={[styles.BtnText, this.state.lectureType === 'date' && styles.BtnTextActived]}>日期</Text>
                    </TouchableHighlight>
                    <TouchableHighlight onPress={()=>this.fetchData('view')} style={[styles.navigateBtn, this.state.lectureType === 'view' && styles.BtnActived]}>
                        <Text style={[styles.BtnText, this.state.lectureType === 'view' && styles.BtnTextActived]}>观看</Text>
                    </TouchableHighlight>
                    <TouchableHighlight onPress={()=>this.fetchData('like')} style={[styles.navigateBtn, this.state.lectureType === 'like' && styles.BtnActived]}>
                        <Text style={[styles.BtnText, this.state.lectureType === 'like' && styles.BtnTextActived]}>喜欢</Text>
                    </TouchableHighlight>
                </View>
                <View style={styles.lecturerContainer}>
                    {this.state.loadingState ?
                        <View style={global_style.loadingContainer}>
                            <Spinner
                                isVisible={true}
                                size={50}
                                type='ThreeBounce'
                                color='#cc3434'
                            />
                        </View>
                        : this.state.albumList.data ?
                            this.renderList()
                            :
                            <TouchableHighlight style={global_style.hintContainer} onPress={ () => this.fetchData }>
                                <Text style={global_style.hintText}>网络异常，请点击重新加载</Text>
                            </TouchableHighlight>
                    }
                </View>
            </View>
        );
    }
}
