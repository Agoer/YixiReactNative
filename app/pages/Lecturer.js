import React, { Component } from 'react';
import {
    View,
    Text,
    Image,
    StyleSheet,
    ScrollView,
    Dimensions,
    TouchableHighlight
} from 'react-native';
import Spinner from 'react-native-spinkit';
import ScrollableTabView, { ScrollableTabBar, } from 'react-native-scrollable-tab-view';

import global_style from '../global_style';
import {fetchTagList, fetchLecturerList} from '../api';

const ScreenWidth = Dimensions.get('window').width;

const styles = StyleSheet.create({
  container : {
     flex: 1,
     backgroundColor : "#fff"
  },
  lecturerContainer : {
      marginTop : 80,
      flex : 1,
      backgroundColor : "#fff"
  },
  barTextStyle : {
      fontWeight : '100',
      fontSize : 14
  },
  underlineStyle : {
      borderTopLeftRadius : 2,
      borderTopRightRadius : 2,
  },
  LecturesContainer : {
      flexDirection : 'row',
      flexWrap : 'wrap',
      backgroundColor : '#efefef'
  },
  lecturerItem : {
      width : ScreenWidth / 3,
      height : 140,
      alignItems : 'center',
      justifyContent : 'center',
  },
  ItemAvatar : {
      height : 60,
      width : 60,
      borderRadius : 30,
      borderWidth : 2,
      borderColor : '#fff',
      shadowOffset : {
         width : 0,
         height : 1,
      },
      shadowRadius: 3,
      shadowOpacity : 1,
      shadowColor : '#ccc',
      alignItems : 'center',
      justifyContent : 'center',
  },
  AvatarPic : {
      height : 56,
      width : 56,
      borderRadius : 28,
      resizeMode : 'contain',
  },
  ItemNickname : {
     marginTop : 12,
     fontSize : 14,
     fontWeight : '100',
     color : '#666'
  }
})

export default class Lecturer extends Component{a
  constructor(props) {
    super(props);
    this.state = {
       lecturerList : [],
       tagList : [],
       tagIDList : [],
       lecturerLoadingState : true,
       tagLoadingState : true,
    }
    this.setState = this.setState.bind(this);
    this.fetcTagData = this.fetcTagData.bind(this);
    this.fetchLecturerData = this.fetchLecturerData.bind(this);
    this.renderLectures = this.renderLectures.bind(this);
  }
  async fetcTagData(initFlag = false) {
    this.setState({
      tagLoadingState : true,
    })
    let jsonData = await fetchTagList();
    let tagIdArr = jsonData.data.map( ele => ele.id);
    this.setState({
       tagList : jsonData,
       tagIDList : [tagIdArr[0]],
       tagLoadingState : false
    })
    if( initFlag ){ this.fetchLecturerData(tagIdArr[0]);}
  }
  async fetchLecturerData(tagId) {
    let jsonData = await fetchLecturerList(tagId);
    this.setState({
        tagIDList : [...new Set([...this.state.tagIDList,tagId])],
        lecturerList : [...this.state.lecturerList, {cat_id: tagId, data : jsonData.data}],
        lecturerLoadingState : false
    })
    return jsonData;
  }
  componentDidMount() {
     this.fetcTagData(true);
  }
  renderLectures(tagId){
      let lecturerList = this.state.lecturerList.find( ele => {
        return ele.cat_id == tagId
      });
      return lecturerList && lecturerList.data.map( (ele, ind) => {
            return <TouchableHighlight key={`lecturer-${ind}`} onPress={ () => this.props.navigation.navigate('Detail',{lecturesId : ele.lectures_with_cover[0].id }) }>
              <View style={styles.lecturerItem}>
                  <View style={styles.ItemAvatar}>
                    <Image source={{ url : ele.pic}} style={styles.AvatarPic} />
                  </View>
                  <Text style={styles.ItemNickname}>{ele.nickname}</Text>
              </View>
            </TouchableHighlight>
          })
  }
  render() {
     return (
        <View style={styles.container}>
          <View style={styles.lecturerContainer}>
          {this.state.tagLoadingState ?
                 <View style={global_style.loadingContainer}>
                    <Spinner isVisible={true}
                             size={50}
                             type='ThreeBounce'
                             color='#cc3434'/>
                 </View>
            : this.state.tagList.data ?
                <ScrollableTabView
                      tabBarActiveTextColor="#cc3434"
                      tabBarTextStyle = {styles.barTextStyle}
                      tabBarUnderlineStyle = { styles.underlineStyle}
                      renderTabBar={() => <ScrollableTabBar />}
                      onChangeTab={(obj) => { this.fetchLecturerData(obj.ref.props.tagId)} }>
                       {this.state.tagList.data.map( (ele, ind) =>
                          <ScrollView tabLabel={ele.name} key={`tag-${ele.id}`} tagId={ele.id} style={styles.scrollContainer}>
                              {this.state.tagIDList.indexOf(ele.id) != -1 ?
                                    <View style={styles.LecturesContainer}>
                                      {this.renderLectures(ele.id)}
                                    </View>
                                  : <View style={global_style.loadingContainer}>
                                      <Spinner isVisible={true}
                                                  size={50}
                                                  type='ThreeBounce'
                                                  color='#cc3434'/>
                                  </View>
                              }
                          </ScrollView>
                       )}
                    </ScrollableTabView>
            : <TouchableHighlight style={global_style.hintContainer} onPress={ () => this.fetchData }>
                 <Text style={global_style.hintText}>网络异常，请点击重新加载</Text>
              </TouchableHighlight>
          }
          </View>
        </View>
     )
  }
}
