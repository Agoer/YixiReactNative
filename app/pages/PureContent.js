import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
    Text,
    StyleSheet,
    ScrollView,
} from 'react-native';

export default class PureContent extends Component{
    static propTypes = {
        navigation: PropTypes.any
    };
    constructor(props) {
        super(props);
    }
    render() {
        let { params } = this.props.navigation.state;
        return (
            <ScrollView style={styles.container}>
                <Text style={styles.purecontent}>{params.title}</Text>
                <Text style={styles.purecontent}>{params.author}</Text>
                <Text style={styles.purecontent}>{params.content}</Text>
            </ScrollView>
        );
    }
}

const styles = StyleSheet.create({
    container : {
        flex : 1,
        marginTop : 80,
        paddingBottom : 140,
        paddingHorizontal : 12
    },
    purecontent : {
        fontSize : 20,
        lineHeight : 28,
        fontWeight : '200',
        marginBottom : 22,
    }
});
