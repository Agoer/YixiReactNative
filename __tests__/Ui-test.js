import 'react-native';
import React from 'react';
import Home from '../app/pages/Home';
import Detail from '../app/pages/Detail';
import Lecture from '../app/pages/Lecture';
import Lecturer from '../app/pages/Lecturer';
import Record from '../app/pages/Record';
import PureContent from '../app/pages/PureContent';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';


test('renders correctly', () => {
  const tree = renderer.create(
    <Home />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly', () => {
  const tree = renderer.create(
    <Lecture />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly', () => {
  const tree = renderer.create(
    <Lecturer />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly', () => {
  const tree = renderer.create(
    <Record />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
