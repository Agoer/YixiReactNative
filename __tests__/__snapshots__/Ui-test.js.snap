// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <RNSpinkit
      color="#cc3434"
      size={50}
      style={
        Array [
          Object {
            "height": 50,
            "width": 50,
          },
          undefined,
        ]
      }
      type="ThreeBounce"
    />
  </View>
</View>
`;

exports[`renders correctly 2`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "flex-end",
        "position": "absolute",
        "right": 12,
        "top": 0,
        "width": 240,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ddd",
              "borderRadius": 4,
              "height": 28,
              "justifyContent": "center",
              "marginLeft": 10,
              "marginTop": 28,
              "width": 48,
            },
            Object {
              "backgroundColor": "#cc3434",
            },
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#333",
              "fontSize": 14,
              "fontWeight": "100",
            },
            Object {
              "color": "#fff",
            },
          ]
        }
      >
        日期
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ddd",
              "borderRadius": 4,
              "height": 28,
              "justifyContent": "center",
              "marginLeft": 10,
              "marginTop": 28,
              "width": 48,
            },
            false,
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#333",
              "fontSize": 14,
              "fontWeight": "100",
            },
            false,
          ]
        }
      >
        观看
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ddd",
              "borderRadius": 4,
              "height": 28,
              "justifyContent": "center",
              "marginLeft": 10,
              "marginTop": 28,
              "width": 48,
            },
            false,
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#333",
              "fontSize": 14,
              "fontWeight": "100",
            },
            false,
          ]
        }
      >
        喜欢
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "marginTop": 80,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <RNSpinkit
        color="#cc3434"
        size={50}
        style={
          Array [
            Object {
              "height": 50,
              "width": 50,
            },
            undefined,
          ]
        }
        type="ThreeBounce"
      />
    </View>
  </View>
</View>
`;

exports[`renders correctly 3`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "marginTop": 80,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <RNSpinkit
        color="#cc3434"
        size={50}
        style={
          Array [
            Object {
              "height": 50,
              "width": 50,
            },
            undefined,
          ]
        }
        type="ThreeBounce"
      />
    </View>
  </View>
</View>
`;

exports[`renders correctly 4`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <RNSpinkit
      color="#cc3434"
      size={50}
      style={
        Array [
          Object {
            "height": 50,
            "width": 50,
          },
          undefined,
        ]
      }
      type="ThreeBounce"
    />
  </View>
</View>
`;
